pkg/plugin/query.go

```diff
@@ -74,10 +59,40 @@ func (q *Query) getQueryURL(rawURL string, queryParams string) (string, error) {
 		return "", fmt.Errorf("expression can't be blank")
 	}
 
+	var u *url.URL
+	var values url.Values
+
 	if q.Instant {
-		return q.queryInstantURL(expr, step, params), nil
+		u, err = newURL(rawURL, instantQueryPath, false)
```

Comment:
i would suggest using two different functions for the queries. 
This will help you better test the behavior for two different requests.